nikolay_shapurov/graph.cpp

```diff
@@ -0,0 +1,40 @@
+#include "graph.hpp"
+#include <assert.h>
+
+void Graph::add_vertex() {
+  vertices_.emplace_back(get_new_vertex_id());
+}
+
+void Graph::add_edge(const VertexId& from_vertex_id,
+                     const VertexId& to_vertex_id) {
+  assert((from_vertex_id > -1) && "The from_vertex_id should be more then -1.");
+  assert((to_vertex_id > -1) && "The to_vertex_id should be more then -1.");
```

Comment:
1) Это уже немного оверкилл, достаточно будет проверить что вершина существует. По-идее, при создании новых вершин мы не передаем `id` извне, а генерируем их внутри графа, что должно работать правильно. Или же нужно писать валидацию генератора `id`
2) Немного не логично, что ты использовал в этой валидации `magic numbers`
```suggestion
  assert((from_vertex_id >= 0) && "The from_vertex_id should be greater or equal to 0.");
  assert((to_vertex_id >= 0) && "The to_vertex_id should be greater or equal to 0.");
```
0 - дефолтное значение, и такая запись более рациональная